client/SDL/SDL3/sdl_touch.cpp

```diff
@@ -70,20 +70,18 @@ BOOL sdl_scale_coordinates(SdlContext* sdl, Uint32 windowId, INT32* px, INT32* p
 		break;
 	}
 
-	if (freerdp_settings_get_bool(sdl->context()->settings, FreeRDP_SmartSizing))
+	if (!fromLocalToRDP)
```

Comment:
well, does it work in all cases now?
e.g.

* smart scaling
* high dpi
* normal connection

Reply:
You're correct. I just ran a quick test and found that it's pretty broken and only works with `+dynamic-resolution`, which is the case I tested before. :joy:

with gnome scaling factor set to 1.5
- `+dynamic-resolution`: fine
- `/smart-sizing:1500x800`: misaligned mouse event and mismatched drawing area with window size
- neither: same as above

I'll look into it tomorrow.

Reply:
It's pretty tricky. When HiDPI is enabled, window size in pixel may not be `FreeRDP_DesktopWidth` x `FreeRDP_DesktopHeight` at init. Plus it would change if you move window to another monitor with different scale factor. This breaks mouse event remapping and causes mismatch between window size (in pixel) and gdi buffer size. Would it be a good idea to send resize request to server even if `+dynamic-resolution` is not enabled?

Reply:
@jpy794 since you can not assume that is supported by the server (think very old windows server and alternate implementations) no.